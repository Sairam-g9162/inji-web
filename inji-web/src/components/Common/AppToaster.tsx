import React from "react";
import { ToastContainer, toast } from "react-toastify";
import { useSelector } from "react-redux";
import { RootState } from "../../types/redux";
import { isRTL } from "../../utils/i18n";

const CustomToast = ({ message }: { message: string }) => (
    <div data-testid="custom-toast">{message}</div>
);

export const AppToaster: React.FC = () => {
    const language = useSelector((state: RootState) => state.common.language);
    const positionClass = isRTL(language) ? "top-left" : "top-right";

    // Example of triggering a toast
    React.useEffect(() => {
        toast(<CustomToast message="Test message" />);
    }, []);

    return (
        <div className={`toast-container ${positionClass}`} data-testid="toast-container">
            <ToastContainer
                position={isRTL(language) ? "top-left" : "top-right"}
                autoClose={1000}
                hideProgressBar
                newestOnTop
                closeOnClick
                rtl={isRTL(language)}
                icon={<React.Fragment />}
                pauseOnFocusLoss
                draggable
                pauseOnHover
                style={{ width: '400px' }}
                theme="colored"
            />
        </div>
    );
};
