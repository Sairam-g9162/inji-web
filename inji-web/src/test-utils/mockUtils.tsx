import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

const renderWithProvider = (ui: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {ui}
            </Router>
        </Provider>,
        options
    );
};

export * from '@testing-library/react';
export { renderWithProvider };
