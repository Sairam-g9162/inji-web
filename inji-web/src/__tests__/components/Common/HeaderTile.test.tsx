import React from 'react';
import {  screen } from '@testing-library/react';
import { HeaderTile } from "../../../components/Common/HeaderTile";
import { renderWithProvider } from '../../../test-utils/mockUtils';


describe("HeaderTile Layout Tests", () => {
    test('check the presence of the container', () => {
        const {asFragment} = renderWithProvider(<HeaderTile content={"No Issuers Found"} subContent={"No Issuers Found"} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
describe("HeaderTile Functionality Tests", () => {
    test('check if content is rendered properly', () => {
        renderWithProvider(<HeaderTile content={"No Issuers Found"} subContent={"No Issuers Found"} />);
        expect(screen.getByTestId("HeaderTile-Text")).toHaveTextContent("No Issuers Found");
    });

    test('check if sub-content is rendered properly', () => {
        renderWithProvider(<HeaderTile content={"No Issuers Found"} subContent={"No Issuers Found"} />);
        expect(screen.getByTestId("HeaderTile-Text-subContent")).toHaveTextContent("No Issuers Found");
    });
});
