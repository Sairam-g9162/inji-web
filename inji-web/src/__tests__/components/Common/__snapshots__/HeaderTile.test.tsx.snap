// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTile Layout Tests check the presence of the container 1`] = `
<DocumentFragment>
  <div
    class="font-bold text-iw-title mt-8 text-xl flex"
    data-testid="HeaderTile-Text"
  >
    No Issuers Found
  </div>
  <div
    class="text-iw-subTitle text-md mb-8 mt-2 flex"
    data-testid="HeaderTile-Text-subContent"
  >
    No Issuers Found
  </div>
</DocumentFragment>
`;
